### PR TITLE
Spotless with custom formatting rules for the Java client

### PR DESCRIPTION
**Pull request by mikaelm12**
I created a custom rule set that covers our conventions. I like this more than the google java formatter because of their apparently intentionally unchangeable line length limit's and their 2 space indentations.

Issue: https://github.com/aspnet/SignalR/issues/2676

### CODE REVIEW

File: clients/java/signalr/build.gradle [hunk 1]
```diff
@@ -20,6 +21,22 @@ dependencies {
...
 }
 
+spotless {
+    java {
+        licenseHeader '// Copyright (c) .NET Foundation. All rights reserved.\n' +
+        '// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.\n' // License header
+
+        importOrder 'java', 'javax', 'org', 'com', 'com.diffplug', ''	// A sequence of package names
+
+        replace      'Not enough space after if', 'if(', 'if ('
```
Comment (BrennanConroy) [inline clients/java/signalr/build.gradle]: for loops and while loops too

File: clients/java/signalr/build.gradle [hunk 2]
```diff
@@ -20,6 +21,22 @@ dependencies {
     implementation 'com.google.code.gson:gson:2.8.5'
 }
 
+spotless {
+    java {
+        licenseHeader '// Copyright (c) .NET Foundation. All rights reserved.\n' +
+        '// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.\n' // License header
```
Comment (BrennanConroy) [inline clients/java/signalr/build.gradle]: Add an extra newline

File: clients/java/signalr/build.gradle [hunk 3]
```diff
@@ -20,6 +21,22 @@ dependencies {
...
+
+        importOrder 'java', 'javax', 'org', 'com', 'com.diffplug', ''	// A sequence of package names
+
+        replace      'Not enough space after if', 'if(', 'if ('
+        replace      'Not enough space after )', '){', ') {'
+        replaceRegex 'Too much space after if', 'if +\\(', 'if ('
+        trimTrailingWhitespace()
```
Comment (BrennanConroy) [inline clients/java/signalr/build.gradle]: Couple rules that would be nice but might be difficult to do:
* no spaces before a comma (not 100% sure about this one)
* single space around all the comparison operators (sounds hard)

File: clients/java/signalr/build.gradle (hunk 1 above)
Comment (anurse) [reply clients/java/signalr/build.gradle]: This is my concern about custom rules... 😟

File: clients/java/signalr/build.gradle (hunk 3 above)
Comment (anurse) [reply clients/java/signalr/build.gradle]: It's disappointing that Spotless doesn't have a way to configure specific rulesets... I'd hoped we could still use standard rules and just turn some of them off.

File: clients/java/signalr/build.gradle (hunk 1 above)
Comment (BrennanConroy) [reply clients/java/signalr/build.gradle]: And `switch`...

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java [hunk 4]
```diff
@@ -163,7 +164,7 @@ private void stop(String errorMessage) {
             return;
         }
 
-        if(errorMessage != null) {
+        if (errorMessage != null) {
             logger.log(LogLevel.Error , "HubConnection disconnected with an error %s.", errorMessage);
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java]: Here is an example of the `,` problem

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java [hunk 5]
```diff
@@ -24,15 +25,15 @@ public WebSocketTransport(String url, Logger logger) throws URISyntaxException {
...
         return url;
     }
 
     private URI formatUrl(String url) throws URISyntaxException {
-        if(url.startsWith(HTTPS)){
+        if (url.startsWith(HTTPS)) {
             url = WSS + url.substring(HTTPS.length());
         }
-        else if(url.startsWith(HTTP)){
+        else if (url.startsWith(HTTP)) {
```
Comment (BrennanConroy) [inline clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java]: `else if` and `else` are supposed to be on the same line as `}`

**Comment by mikaelm12** [comment]
Closing in favor of https://github.com/aspnet/SignalR/pull/2924

File: clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java (hunk 5 above)
Comment (mikaelm12) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java]: I'll apply this to the other pr

Comment (mikaelm12) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java]: https://github.com/aspnet/SignalR/pull/2924

Comment (BrennanConroy) [reply clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java]: :/

**Pull request closed without merging**